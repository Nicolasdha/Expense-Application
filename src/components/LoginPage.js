import React from 'react'


export const LoginPage = () => (

    <div>
        <p>LOGIN PAGE</p>
        <button>Login</button>
    </div>

);
 