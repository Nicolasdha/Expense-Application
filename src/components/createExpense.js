import React from 'react';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';
import ExpenseForm from './expenseForm'



export class CreateExpensePage extends React.Component {

    onSubmit = (expense) => { 
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };

    render(){
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm onSubmit = {this.onSubmit}/>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
})


export default connect(undefined, mapDispatchToProps)(CreateExpensePage)


// const CreateExpensePage = (props) => (
//     <div>
//         <h1>Add Expense</h1>
//         <ExpenseForm
//             onSubmit = {(expense) => { 
//                 props.onSubmit(expense);
//                 props.history.push('/');
//             }}
//         />
//     </div>
// );




// const CreateExpensePage = (props) => (
//     <div>
//         <h1>Add Expense</h1>
//         <ExpenseForm
//             onSubmit = {(expense) => { 
//                 props.dispatch(addExpense(expense))
//                 props.history.push('/')
//             }}
//         />
//     </div>
// );

// export default connect()(CreateExpensePage)