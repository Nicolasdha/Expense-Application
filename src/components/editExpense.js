import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal'
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './expenseForm'




export class EditExpensePage extends React.Component {
    state = {
        modalState: undefined
      };

    onSubmit = (expense) =>{ 
            this.props.startEditExpense(this.props.match.id, expense)
            this.props.history.push('/')
    };

    openModal = () => {
        this.setState(()=>({ modalState: true }
            ));
     };
     
     closeModal = () => {
         this.setState(()=>({ modalState: undefined }))
     }

     onRemove = () =>{
        this.setState(()=>({ modalState: undefined }
        ));
        this.props.startRemoveExpense( {id: this.props.match.id} )
        this.props.history.push('/')
      }

    render() {

        return (
            <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Edit Expense</h1>
                </div>
            </div>
                <div className="content-container">
                    <ExpenseForm
                        match = {this.props.match}
                        onSubmit = {this.onSubmit}
                        />
                    <button className="button button--secondary" onClick={this.openModal}>Remove Expense</button>
                    <Modal
                        isOpen={!!this.state.modalState} 
                        contentLabel="Are you sure you want to remove this expense?"
                        appElement={document.getElementById('app')}
                        onRequestClose={this.closeModal}
                        closeTimeoutMS={200} 
                        className="modal"
                    >
                       <div className="modal__titleContainer">
                            <h3 className="modal__title">Remove expense?</h3>
                       </div>
                        <div className="modal__buttonContainer">
                            <button className="button modal--button" onClick={this.onRemove}>Yes</button>
                            <button className="button modal--button" onClick={this.closeModal}>No</button>
                        </div>

                    </Modal>
                </div>
            </div>
        );
    };
};


const mapStoreToProps = (state, props) =>({
    match: state.expenses.find((each)=> each.id === props.match.params.id),
});



const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (id)=> dispatch(startRemoveExpense(id))
});



export default connect(mapStoreToProps, mapDispatchToProps)(EditExpensePage);




// const EditExpensePage = (props) => {
//     const match = props.expenses.find((each)=> each.id === props.match.params.id)
//     return (
//     <div>
//         <ExpenseForm
//             match = {match}
//             onSubmit = {(expense) =>{ 
//                 props.dispatch(startEditExpense(match.id, expense))
//                 props.history.push('/')
//             }}
//             />
//         <button
//             onClick={() => {
//                 const removeConfirmation = window.prompt('Are you sure you want to remove? Type Y to confirm')
//                 if(removeConfirmation === 'yes'.toLowerCase() || 'y'.toLowerCase()){
//                     props.dispatch(startRemoveExpense( {id: match.id} ))
//                     props.history.push('/')
//                 }
//              }}
//         >Remove</button>
//     </div>
//     );
// };
