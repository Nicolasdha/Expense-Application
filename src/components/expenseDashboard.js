import React from 'react';


import ExpenseList from './expenseList';
import ExpenseListFilters from './expenseListFilters';
import ExpensesSummary from './expensesSummary';


const ExpenseDashboardPage = () => {
   
return ( 
    <div>
        {<ExpenseListFilters/>}
        {<ExpensesSummary/>}
        {<ExpenseList/>}
    </div>
    );
};


export default ExpenseDashboardPage;
