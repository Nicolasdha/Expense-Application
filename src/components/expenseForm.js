import React from 'react';
import moment from'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/initialize';



class ExpenseForm extends React.Component {

    constructor(props){ 
      super(props);
        this.state = {
            description: props.match ? props.match.description : '',
            note: props.match ? props.match.note : '',
            amount: props.match ? (props.match.amount / 100).toString() : '',
            createdAt: props.match ? moment(props.match.createdAt) : moment(),
            calandarFocused: false,
            error: '',
        } 
    }

    
    onDescriptionChange = (e) => {
        const description = e.target.value
        this.setState(() =>({ description }))
    };

    onNoteChange = (e) =>{
        const note = e.target.value
        this.setState(() =>({ note }))
    };

    onAmountChange = (e) =>{
        const amount = e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/g)){
            this.setState(()=>({ amount }))
        }
    };

    onDateChange = (createdAt) =>{
        if(createdAt){
            this.setState(()=> ({ createdAt }))
        }
    };

    onFocusChange = ( { focused } ) => {
        this.setState(()=>({ calandarFocused: focused}))
    };

    onSubmit = (e) =>{
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(()=> ({error: 'Please provide Description and Amount'}));
        } else{
            this.setState(()=> ({error: ''}));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                note: this.state.note,
                createdAt: this.state.createdAt.valueOf()
            })
        }
    }

    render(){
        return (
            <div>
              {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text" 
                        placeholder="Description" 
                        value = {this.state.description} 
                        onChange = {this.onDescriptionChange}
                        autoFocus
                    />
                    <input 
                        type="text" 
                        placeholder="Amount"
                        value = {this.state.amount}
                        onChange = {this.onAmountChange}
                    />
                    <SingleDatePicker 
                        date = {this.state.createdAt}
                        onDateChange = {this.onDateChange}
                        focused = {this.state.calandarFocused}
                        onFocusChange = {this.onFocusChange}
                        numberOfMonths = {1}
                        isOutsideRange = {(day)=>false}
                    />
                    <textarea 
                        placeholder="Add a note (optional)"
                        value={this.state.note}
                        onChange = {this.onNoteChange}
                    ></textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        );
    };
};

export default ExpenseForm;
