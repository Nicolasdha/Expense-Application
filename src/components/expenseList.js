import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './expenseListItem';
import getVisibleExpenses from '../selectors/expenses';



export const ExpenseList = (props) => {
    
    return (
        <div>
            {
                props.expenses.length === 0 ? (
                    <p>Please add expense!</p>) : (
                        props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense}/>) 
                    )
            }
        </div>
)};

const mapStoreToProps = (state) =>{
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStoreToProps)(ExpenseList);
