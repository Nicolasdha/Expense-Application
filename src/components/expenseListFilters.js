import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';
import { sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

import { DateRangePicker } from 'react-dates';
import 'react-dates/initialize';



export class ExpenseListFilters extends React.Component {
    state = {
        calanderFocused: null,
    };

    onDatesChange =({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    onFocusChange = (newCalanderFocused) => {
        this.setState(() => ({ calanderFocused: newCalanderFocused }))
    }

    onTextChange = (e) =>{
        this.props.setTextFilter(e.target.value)
    }

    onSortChange = (e) => {
        if(e.target.value==="date"){
            this.props.sortByDate();
        } else if (e.target.value==="amount"){
            this.props.sortByAmount();
        }
    }

    render() {
        return (
            <div>
            <h4>Filter</h4>
            <input
                type='text'
                value={this.props.filters.text}
                onChange={this.onTextChange}
            />
            <label htmlFor="sortBy" >Sort By</label>
                <select 
                    name="sortBy" 
                    value={this.props.filters.sortBy} 
                    onChange={this.onSortChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
    
            <DateRangePicker 
                startDate={this.props.filters.startDate} 
                startDateId="startDateId"
                endDate={this.props.filters.endDate}
                endDateId="endDateId"
                onDatesChange={this.onDatesChange} 
                focusedInput={this.state.calanderFocused} 
                onFocusChange={this.onFocusChange} 
                isOutsideRange = {(day)=>false}
                numberOfMonths = {1}
                showClearDates= {true}
                showDefaultInputIcon
            />
        </div>
        )
    }
}
   


const mapStoreToProps = (state) =>{
    return {
        filters: state.filters
    }
};

const mapDispatchToProps = (dispatch) =>({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (textValue) => dispatch(setTextFilter(textValue)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount())

})

export default connect(mapStoreToProps, mapDispatchToProps)(ExpenseListFilters);
