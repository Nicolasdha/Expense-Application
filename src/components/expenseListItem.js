import React from 'react';
import moment from 'moment'
import numeral from 'numeral'
import { Link } from 'react-router-dom';



const ExpenseListItem =  ({ description, amount, createdAt, note, id }) => {
    let expenseAmountForTotal = amount
    return (
            <Link className="list-item" to={`/edit/${id}`}>
            <div>
                <h3 className="list-item__title">{description}</h3>
                <span className="list-item__sub-title">{createdAt && <p>{moment(createdAt).format("dddd, MMMM Do YYYY")}</p>}</span>
            </div>
                <h3 className="list-item__data">
                    {numeral(amount / 100).format('$0,0.00')}
                </h3>
            </Link>
)};

// {note && <p> Note: {note}</p>}


export default ExpenseListItem;

