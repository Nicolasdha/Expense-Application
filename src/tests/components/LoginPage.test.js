import React from 'react';
import {LoginPage} from '../../components/LoginPage'
import { shallow } from 'enzyme';
 
test('should render login page correctly', () =>{
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
})