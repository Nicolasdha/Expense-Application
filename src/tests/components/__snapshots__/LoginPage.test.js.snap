// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page correctly 1`] = `
<div>
  <p>
    LOGIN PAGE
  </p>
  <button>
    Login
  </button>
</div>
`;
